### PR TITLE
Fixes #13432: typo in #11172 causing notations mentioning a coercion not being used

### LINKED ISSUE #13432: Coercion masks notation
```coq
Record word := { rep: Type }.

Coercion rep : word >-> Sortclass.

Axiom myword: word.

Axiom foo: myword.

Notation "'(_' 'BitVec' '32)'" := (rep myword).

About foo.
```

prints

```
foo : myword
```

Desired behavior: It should print

```
foo : (_ BitVec 32)
```

As a workaround, if I do `Set Printing Coercions.`, I get the desired output, but I'm not sure if I'd always want to print all coercions to make notations work.

Not sure if this is a bug or a feature though...

Coq version: 8.12.0

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** bug fix 

This fixes a regression introduced in #11172 (an `=` that became a `>` while it should have been a `>=`).

This hopefully puts an end to the series #8890, #11090, #11172 of PRs on notations and coercions with the following basic rule: if `f is a coercion on `t` in `f params t`, a notation is used prioritary to skipping the coercion as soon as the notation extends at least over the whole subterm `f param t`. If `f` is to funclass and the term is `f params t u`, the notation is also used, and `u` is applied afterwards.

Fixes / closes #13432.

- [X] Added / updated test-suite
- [ ] Entry added in the changelog

### CODE REVIEW

**Comment by gares** [comment]
Why is this in the 8.13 milestone? Why is this fix 8.13 specific?

**Review by ejgallego — APPROVED** [review]

**Comment by ejgallego** [comment]
> Why is this in the 8.13 milestone? Why is this fix 8.13 specific?

The bug was introduced in 8.12, thus it seems a good idea to ship a fix in 8.13? I'll wait for feedback before merge.

**Comment by gares** [comment]
ok

**Comment by ejgallego** [comment]
@coqbot: merge now

**Comment by ejgallego** [comment]
@coqbot: merge now

**Comment by CohenCyril** [comment]
This PR seems to fix an issue that breaks some notations displays in Coq 8.12 cf https://github.com/math-comp/math-comp/pull/674
Could you please backport it to Coq 8.12 ASAP?

**Pull request merged**
